src/power_grid_model/core/power_grid_model.py

```diff
@@ -6,7 +6,7 @@
 Main power grid model class
 """
 from enum import IntEnum
-from typing import Dict, List, Optional, Set, Type, Union
+from typing import Optional, Set, Type
```

Comment:
please double-check that all cases of List, Tuple, Set, Dict have been removed
```suggestion
from typing import Optional, Type
```